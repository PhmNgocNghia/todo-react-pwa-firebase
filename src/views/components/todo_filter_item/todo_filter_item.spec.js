
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import TodoFilterItem from './index';

describe('Todo Search', function() {
  // Render
  it('should render button with active class when current state equal state', function () {
    const props = {
      state: {
        name: 'A',
        text: 'a'
      },
      currentState: 'A'
    }

    let shallowTodoFilterItem = shallow(<TodoFilterItem {...props} />)
    expect(shallowTodoFilterItem.find('button').hasClass('active')).toBeTruthy()
  })

  it('should not render button without active class when current state not equal state', function () {
    const props = {
      state: {
        name: 'A',
        text: 'a'
      },
      currentState: 'B'
    }

    let shallowTodoFilterItem = shallow(<TodoFilterItem {...props} />)
    expect(shallowTodoFilterItem.find('button').hasClass('active')).toBeFalsy()
  })

  // Event
  it('should run onclick when press', function () {
    const props = {
      state: {
        name: 'A',
        text: 'a'
      },
      currentState: 'B',
      onClick: jest.fn()
    }

    let shallowTodoFilterItem = shallow(<TodoFilterItem {...props} />)
    shallowTodoFilterItem.find('button').simulate('click')
    expect(props.onClick).toHaveBeenCalled()
  })

  // Snapshoot
  it('should match snapshoot when state equal current state', function () {
    const props = {
      state: {
        name: 'A',
        text: 'a'
      },
      currentState: 'A'
    }

    let shallowTodoFilterItem = shallow(<TodoFilterItem {...props} />)
    expect(shallowTodoFilterItem).toMatchSnapshot()
  })

  it('should match snapshoot when state not equal current state', function () {
    const props = {
      state: {
        name: 'A',
        text: 'a'
      },
      currentState: 'B'
    }

    let shallowTodoFilterItem = shallow(<TodoFilterItem {...props} />)
    expect(toJson(shallowTodoFilterItem)).toMatchSnapshot()
  })
})