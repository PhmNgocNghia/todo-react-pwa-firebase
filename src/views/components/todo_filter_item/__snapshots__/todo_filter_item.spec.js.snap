// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Search should match snapshoot when state equal current state 1`] = `
<styled.div>
  <button
    className="btn btn-link px-2 py-0 my-1 active"
    id="A"
    onClick={[Function]}
  >
    a
  </button>
  <span>
    |
  </span>
</styled.div>
`;

exports[`Todo Search should match snapshoot when state not equal current state 1`] = `
<styled.div>
  <button
    className="btn btn-link px-2 py-0 my-1 "
    id="A"
    onClick={[Function]}
  >
    a
  </button>
  <span>
    |
  </span>
</styled.div>
`;
